### ISSUE TITLE
Generic type aliases show two inlay hints

### ISSUE THREAD

**Issue by Jammf** [microsoft/pylance-release#6427]
Generic type aliases show two inlay hints that each function independently, with no visual separation between them. I'm not sure if it's just restricted to type aliases with generics, but I first noticed it happening when hints from numpy functions such as `np.array(3)` would be things like `Anyndarray[Any, dtype[Any]]`. I tried to reduce it down to a minimal example from there.

<h3> Environment data </h3>

-   Language Server version: 2024.9.101
-   OS and version: darwin arm64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.12.4
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: off

<h3> Code Snippet </h3>

```python
from typing import TypeVar

T = TypeVar("T")
type U = list[T]

def func() -> U: ...
a = func()

def func2() -> U[int]: ...
b = func2()

```

<h3> Expected behavior </h3>

I'd expect only the second inlay hint to be present, for example `a: list` and `b: list[int]`.

<h3> Actual behavior </h3>

<h4>(Previous) Behavior in 2024.8.100:</h4>

<img width="346" alt="image" src="https://github.com/user-attachments/assets/6a394dd9-b69d-4f72-8a25-adf1ee1f1b70">

<h4>(Current) Behavior from 2024.8.101 to 2024.9.101:</h4>

<img width="379" alt="image" src="https://github.com/user-attachments/assets/c9a88630-5406-43c0-a6dd-e27a0e433b2a">

<h3> Logs </h3>

```
XXX
```

**Comment by debonte** [comment]
@StellaHuang95, reassigned to you. Given that this regressed in 2024.8.101, I'm guessing this has something to do with the inlay hint go to def feature.

**Comment by StellaHuang95** [comment]
This issue has been fixed in prerelease version 2024.9.102, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20249102-26-September-2024-PreRelease)
